include/taskolib/GitWrapper.h

```diff
@@ -0,0 +1,148 @@
+/**
+ * \file   GitWrapper.h
+ * \author Sven-Jannik Wöhnert
+ * \date   Created on March 20, 2023
+ * \brief  Wrapper for C-Package libgit2
+ *
+ * \copyright Copyright 2023 Deutsches Elektronen-Synchrotron (DESY), Hamburg
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 2.1 of the license, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+
+#include <git2.h>
+#include <string>
+#include <filesystem>
+#include "taskolib/GitObjectWrapper.h"
+#include <vector>
+
+
+
+
+/**
+ * A class to wrap used methods from C-Library libgit2
+ * 
+ * This class is necessary to handle C-pointers in a way that
+ * C++ programmers using this class do not have to think about them.
+ * 
+ * This class does not reflect the full libgit2 library.
+ * git functions which are not implemented in this class will be
+ * implemented in case of necessity
+*/
+class LibGit
+{
+public:
+
+    /**
+     * Constructor which specifies the root dir of the git repository
+     * \param file_path: path to "sequences"-folder (or customized directory)
+     *
+     */
+    explicit LibGit(std::filesystem::path file_path);
+
+    /// returns member variable, which is the root dir of the git repository
+    std::filesystem::path get_path() const;
+
+    /// returns a C-type pointer to the current loaded repository
+    git_repository* get_repo();
```

Comment:
Why would an end user need this raw pointer?

Reply:
He does not need it, I needed it for the UnitTests